src/resources/views/crud/filters/date_range.blade.php

```diff
@@ -90,31 +142,65 @@ function applyDateRangeFilter{{camel_case($filter->name)}}(start, end) {
 			// mark this filter as active in the navbar-filters
 			if (URI(new_url).hasQuery('{{ $filter->name }}', true)) {
 				$('li[filter-name={{ $filter->name }}]').removeClass('active').addClass('active');
-			}
-			else
-			{
+			} else {
 				$('li[filter-name={{ $filter->name }}]').trigger('filter:clear');
 			}
   		}
 
 		jQuery(document).ready(function($) {
-			var dateRangeInput = $('#daterangepicker-{{ str_slug($filter->name) }}').daterangepicker({
-				timePicker: false,
-		        ranges: {
-		            'Today': [moment().startOf('day'), moment().endOf('day')],
-		            'Yesterday': [moment().subtract(1, 'days'), moment().subtract(1, 'days')],
-		            'Last 7 Days': [moment().subtract(6, 'days'), moment()],
-		            'Last 30 Days': [moment().subtract(29, 'days'), moment()],
-		            'This Month': [moment().startOf('month'), moment().endOf('month')],
-		            'Last Month': [moment().subtract(1, 'month').startOf('month'), moment().subtract(1, 'month').endOf('month')]
-		        },
-				@if ($filter->currentValue)
-		        startDate: moment("{{ $start_date }}"),
-		        endDate: moment("{{ $end_date }}"),
-				@endif
-				alwaysShowCalendars: true,
-				autoUpdateInput: true
-			});
+			var dateRangeInput = $('#daterangepicker-{{ str_slug($filter->name) }}');
+
+            $config = dateRangeInput.data('bs-datepicker');
+
+            $ranges = $config.ranges;
+            $config.ranges = {};
+
+            //the use of eval() is not always recommended, but this use case fits
+            //the string beeing eval'd comes from a trusty source because is always hardcoded (dev or us).
+            for (var key in $ranges) {
+                if ($ranges.hasOwnProperty(key)) {
+                    $config.ranges[key] = eval($ranges[key]);
+                }
+            }
+
+            @if(!$filter->currentValue)
+
+            //this is used mainly to avoid converting valid Carbon dates to other formats.
+
```

Comment:
Ugh. Usually when I see regex I think "_there must be a better way to do it_". I see from the comment that this is "_used mainly to avoid converting valid Carbon dates to other formats_" - but what does that mean? What other formats could the date be converted to? What use case is this for, or what does it try to prevent?